packages/cli/generators/controller/index.js

```diff
@@ -73,3 +148,90 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
     }
   }
 };
+
+/**
+ * Optional prompt to retrieve variables for artifacts.
+ * Not a member of the class because Yeoman will always call Generator
+ * prototype functions! Use .apply or .call with this function!
+ */
+function promptArtifactCrudVars() {
+  const self = this;
+  let modelList = [];
+  let repositoryList = [];
+  return utils
+    .getArtifactList(self.artifactInfo.modelDir, 'model')
+    .then(list => {
+      if (_.isEmpty(list)) {
+        return Promise.reject(
+          new Error(`No models found in ${self.artifactInfo.modelDir}`)
+        );
+      }
+      modelList = list;
+      return utils.getArtifactList(
+        self.artifactInfo.repositoryDir,
+        'repository',
+        true
+      );
+    })
+    .then(list => {
+      if (_.isEmpty(list)) {
+        return Promise.reject(
+          new Error(
+            `No repositories found in ${self.artifactInfo.repositoryDir}`
+          )
+        );
+      }
+      repositoryList = list;
+      return self
+        .prompt([
+          {
+            type: 'list',
+            name: 'modelName',
+            message:
+              'What is the name of the model to use with this CRUD repository?',
+            choices: modelList,
+            when: self.artifactInfo.modelName === undefined,
+            default: modelList[0],
+          },
+          {
+            type: 'list',
+            name: 'repositoryName',
+            message: 'What is the name of your CRUD repository?',
+            choices: repositoryList,
+            when: self.artifactInfo.repositoryName === undefined,
+            default: repositoryList[0],
+          },
+          {
+            type: 'list',
+            name: 'id',
+            message: 'What is the type of your ID?',
+            choices: ['number', 'string', 'object'],
+            when: self.artifactInfo.id === undefined,
+            default: 'number',
+          },
+        ])
+        .then(props => {
+          debug(`props: ${inspect(props)}`);
+          Object.assign(self.artifactInfo, props);
+          // Ensure that the artifact names are valid.
+          [
+            self.artifactInfo.name,
+            self.artifactInfo.modelName,
+            self.artifactInfo.repositoryName,
+          ].forEach(item => {
+            const validationMsg = utils.validateClassName(item);
+            if (typeof validationMsg === 'string')
+              throw new Error(validationMsg);
+            item = utils.toClassName(item);
+          });
+          // Create camel-case names for variables.
+          self.artifactInfo.repositoryNameCamel = utils.camelCase(
+            self.artifactInfo.repositoryName
+          );
+          self.artifactInfo.modelNameCamel = utils.camelCase(
+            self.artifactInfo.modelName
+          );
+          return Promise.resolve(props);
```

Comment:
No need to wrap props into a promise. A normal return value for `then()` will be converted to a promise automatically.